Assets/MixedRealityToolkit.SDK/Features/UX/Scripts/Collections/BaseObjectCollection.cs

```diff
@@ -14,10 +14,17 @@ public abstract class BaseObjectCollection : MonoBehaviour
         /// </summary>
         public Action<BaseObjectCollection> OnCollectionUpdated { get; set; }
 
+        [HideInInspector]
+        [SerializeField]
+        private List<ObjectCollectionNode> nodeList = new List<ObjectCollectionNode>();
```

Comment:
See if you can prevent this from being serielized

Reply:
Curious what the pushback is here? A couple arguments for the change:

- In its current state, the list is inaccessible through `SerializedObject` (custom inspectors).
- The list is of type `ObjectCollectionNode` which are `Serialized`.
- The most common use case for `ObjectCollection` is setting them up at editor time. Pressing 
`Update Collection` generates this list. As a user, I'd expect the list to keep its state.
- The original implementation serialized this field. https://github.com/microsoft/MixedRealityToolkit-Unity/blob/91107545a518aa543fd43376daa9ad420fa03fb9/Assets/HoloToolkit/UX/Scripts/Collections/ObjectCollection.cs#L27
- IMO its a regression (which I take blame since I was the one who brought `ObjectCollection` to MRTK v2) The original PR #2606 had this as a public (serialized) field, which was changed due to PR feedback.